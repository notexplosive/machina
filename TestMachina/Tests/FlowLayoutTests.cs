using ApprovalTests;
using ApprovalTests.Reporters;
using Machina.Data.Layout;
using TestMachina.Utility;
using Xunit;

namespace TestMachina.Tests
{
    [UseReporter(typeof(DiffReporter))]
    public class FlowLayoutTests
    {
        [Fact]
        public void can_create_flow_layout()
        {
            var layout = FlowLayout.FlowParent("root", LayoutSize.Pixels(30, 30), LayoutStyle.Empty,
                LayoutNode.Leaf("itemA", LayoutSize.Pixels(10, 10)),
                LayoutNode.Leaf("itemB", LayoutSize.Pixels(10, 10)),
                LayoutNode.Leaf("itemC", LayoutSize.Pixels(10, 10)),
                LayoutNode.Leaf("itemD", LayoutSize.Pixels(10, 10)),
                LayoutNode.Leaf("itemE", LayoutSize.Pixels(10, 10)),
                LayoutNode.Leaf("itemF", LayoutSize.Pixels(10, 10))
            );

            var result = layout.Bake();
            Approvals.Verify(LayoutNodeUtils.DrawResult(result));
        }
    }
}