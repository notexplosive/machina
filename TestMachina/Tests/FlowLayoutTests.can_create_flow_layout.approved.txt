333333333333333333333333333333
3itemA   33itemB   33itemC   3
3        33        33        3
3        33        33        3
3        33        33        3
3        33        33        3
3        33        33        3
3        33        33        3
3        33        33        3
333333333333333333333333333333
333333333333333333333333333333
3itemD   33itemE   33itemF   3
3        33        33        3
3        33        33        3
3        33        33        3
3        33        33        3
3        33        33        3
3        33        33        3
3        33        33        3
333333333333333333333333333333
1                            1
1                            1
1                            1
1                            1
1                            1
1                            1
1                            1
1                            1
1                            1
111111111111111111111111111111
